### PR TITLE
[SUP-519] Add killswitch for workspace-level notifications

### PR DESCRIPTION
**Pull request by MatthewBemis**
Thurloe supports two classes of email notifications:

`UserNotification`: users can disable these types of notifications at the account-level
`WorkspaceNotification`: users can disable these types of notifications at the workspace-level

We recently added a new email notification for when a user's submission completes, which is a `WorkspaceNotification`. As anticipated, users have given feedback that they also want to be able to disable these notifications across the board, a la `UserNotification`. Thurloe obviously doesn't support that currently, so this adds in some logic to allow for users to disable `WorkspaceNotification`s at both the workspace-level and the account-level.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by MatthewBemis** [comment]
jenkins retest

File: src/main/scala/thurloe/notification/NotificationMonitor.scala [hunk 1]
```diff
@@ -194,12 +194,29 @@ class NotificationMonitorActor(val pollInterval: FiniteDuration,
...
+      //For workspace notifications, there are three tiers of preferences to check:
+      // 1. has the user disabled *all* notifications for their account?
+      //     key format: notifications.off
+      // 2. has the user disabled the notification at the type-level?
+      //     key format: notifications/<notification class name>
+      // 3. has the user disabled the notification for the specified workspace?
+      //     key format: notifications/<notification class name>/<workspace namespace>/<workspace name>
+      case wsNotification: WorkspaceNotification =>
+        val baseKey = s"notifications/${wsNotification.getClass.getSimpleName}"
+        //Check #1
```
Comment (nawatts) [inline src/main/scala/thurloe/notification/NotificationMonitor.scala]: If this check is to be labeled in a comment, should checks 2 and 3 also be labeled?

File: src/main/scala/thurloe/notification/NotificationMonitor.scala [hunk 2]
```diff
@@ -194,12 +194,29 @@ class NotificationMonitorActor(val pollInterval: FiniteDuration,
...
+        booleanLookup(wsNotification.recipientUserId, NotificationMonitor.notificationsOffKey, false) flatMap {
+          case false =>
+            booleanLookup(wsNotification.recipientUserId, notification.key, true) flatMap {
+              case false => Future.successful(false)
+              case true  => booleanLookup(wsNotification.recipientUserId, baseKey, true)
+            }
```
Comment (nawatts) [inline src/main/scala/thurloe/notification/NotificationMonitor.scala]: If I'm reading this right, this checks `notifications/<notification class name>/<workspace namespace>/<workspace name>` before  `notifications/<notification class name>`? I guess it doesn't matter, but I would have expected the checks to follow the order in the comment.

**Review by nawatts — APPROVED** [review]

File: src/main/scala/thurloe/notification/NotificationMonitor.scala (hunk 2 above)
Comment (MatthewBemis) [reply src/main/scala/thurloe/notification/NotificationMonitor.scala]: Whoops yeah, ideally it should follow the order in the comment. I'll update that for sanity

File: src/main/scala/thurloe/notification/NotificationMonitor.scala (hunk 1 above)
Comment (MatthewBemis) [reply src/main/scala/thurloe/notification/NotificationMonitor.scala]: Comment was leftover from me mapping out how this logic should work, I'll just remove this comment

**Comment by MatthewBemis** [comment]
jenkins retest

leo failed to boot

**Review by Ghost-in-a-Jar — APPROVED** [review]

**Pull request merged**